### ISSUE TITLE
[Q] Compatibility with Qt commercial qtcreator 

### ISSUE THREAD

**Issue by sparkleholic** [Pelagicore/meta-pelux#242]
I'm trying to test if I can make app with qtcreator and install via appman to the pelux.
I installed qtcreator(commercial trial) which provides 'automotive suites examples'.

First of all, I configured the pelux toolchain kit as described in [pelux doc](https://pelux.io/software-factory/master/swf-blueprint/docs/articles/sdk/configure-qtcreator.html#)
Then I selected a example in the qtcreator, which named 'hello-world'. 
The sample just  has only qml files without native C/C++ file.
Since the sample consists of qml files only, it doesn't need link step. But generated Makefile by qmake has the following line.
```
$(TARGET):  $(OBJECTS)  
    $(LINK) $(LFLAGS) -o $(TARGET) $(OBJECTS) $(OBJCOMP) $(LIBS)
```
However if I did with pre-existing automotive-suite toolchain included in the qt installer (for intel NUC), the generated Makefile was slight different.
```
hello-world:  $(OBJECTS)  
    
```
Eventually, it fails on 'link' step when I try with pelux toolchain. But with 'intel NUC' automotive-toolchain, there is no issue.

Do you know why different Makefiles are created based on the same qmake project file?

**Comment by Gagi2k** [comment]
Hi,

the difference in the Makefile is probably because you are cross-compiling which very often creates some slightly different Makefiles.

From the description it sounds like you would like to deploy and run an app from within QtCreator to a device which runs PELUX and a appman system-ui e.g. neptune3 or minidesk ?
If that's the case the 'hello-world' example is not really what you are looking for, as it is a whole appman example which deploys a complete new system-ui and it's own apps.

If you just want to deploy an application to an already running system-ui, the appman examples are not a 100% fit, as they are more intended to show how to right a system-ui and use appman's features for this. For deploying an app you might want to try out the new wizard we added to create an appman application:
File -> New File or Project -> Application -> Qt ApplicationManager Project

**Comment by sparkleholic** [comment]
New qt-appMan project seems to be installed well.  Thanks a lot! :+1: 

Do you know how to deal with the error (warning) that qtcreator can not find the QtApplicationManager module as follows?
![image](https://user-images.githubusercontent.com/2949977/51605548-c4bcd600-1f52-11e9-9bde-a5e8e18a5ab7.png)

**Comment by Gagi2k** [comment]
Hi, yes, we are aware of this problem and are trying to fix it with the 5.12.1 release.

The problem is that appman is not shipped as a normal QML plugin and because of that QtCreator cannot find it. In the 5.12.1 release we will ship some code completion files which will help to fix the warning and will provide code completion for all appman types

**Comment by sparkleholic** [comment]
@Gagi2k  Thanks for reply.
I expected the `commercial qtcreator` would provide many features for automotive suites, but I only found there is a way to use `AppMgr` for packing/running auto apps.
And I couldn't find any gui widgets for qtcreator desgine mode, 
Could you let me know if there is a way to enable neptune ui widgets in the qtcreator-designer? 

I just left some [questions](https://forum.qt.io/topic/98863/qt-automotive-suites-examples) on the qt forum about automotive suite examples.

On creating new "appman project" in qtcreator, if I checked "system ui" checkbox, qtcreator fails to launch the app. With unchecking, it seems to work fine.

**Comment by sparkleholic** [comment]
Thanks @Gagi2k .
I have some questions for the qt designer.

By default, in order to view the qml file in the designer view, does the qml plugins which in the qml file need to be loadable on the PC?
If the qml plugin is cross-compiled only, it seems not be shown on the designer view. 
And some qml plugins which are only loadable in the certain environment like wayland.(i.g. ApplicationManagerWindow).  These plugin types seem not be shown on the designer view regardless it is compiled for x86 or target.

If I misunderstood, please let me know.

**Comment by Gagi2k** [comment]
Sorry i don't know, i never used the QML form editor

**Comment by minenko** [comment]
@sparkleholic No, you are fully correct. The "Designer" in Qt Creator was initially not designed for non-desktop environments which could use custom Window types. What surprises more is that the new Qt Design Studio does not consider this either whereas it was made in awareness of embedded systems which might not be based on the standard Window type. I've created a bug report to clarify this: https://bugreports.qt.io/browse/QDS-511
